File: src/go/fiber/internal/models/complex_order_result.go
package models

import "fmt"

// ComplexOrderResult mirrors UserOrderStats in contracts/grpc/benchmark.proto
// and type UserOrderStats in contracts/graphql/schema.graphql, so /db/complex
// returns the same shape over all three protocols.
//
// Field names are camelCase, matching the proto3 JSON mapping of the
// snake_case proto fields. See contracts/rest/canonical-payloads.md.
type ComplexOrderResult struct {
	UserID            int     `json:"userId"`
	UserName          string  `json:"userName"`
	TotalOrders       int64   `json:"totalOrders"`
	TotalValue        float64 `json:"totalValue"`
	AverageOrderValue float64 `json:"averageOrderValue"`
}

// NewComplexOrderResult creates a new ComplexOrderResult instance
func NewComplexOrderResult(
	userID int,
	userName string,
	totalOrders int64,
	totalValue float64,
	averageOrderValue float64,
) *ComplexOrderResult {
	return &ComplexOrderResult{
		UserID:            userID,
		UserName:          userName,
		TotalOrders:       totalOrders,
		TotalValue:        totalValue,
		AverageOrderValue: averageOrderValue,
	}
}

// String returns a string representation of the ComplexOrderResult
func (cor *ComplexOrderResult) String() string {
	return fmt.Sprintf("ComplexOrderResult{UserID: %d, Orders: %d}",
		cor.UserID, cor.TotalOrders)
}

File: src/java/quarkus/src/main/java/com/benchmark/api/model/ComplexOrderResult.java
package com.benchmark.api.model;

import com.fasterxml.jackson.annotation.JsonProperty;

// Mirrors UserOrderStats in contracts/grpc/benchmark.proto. Wire names are
// camelCase, matching the proto3 JSON mapping of the snake_case proto fields.
// See contracts/rest/canonical-payloads.md.
public class ComplexOrderResult {

    @JsonProperty
    public Integer userId;

    @JsonProperty
    public String userName;

    @JsonProperty
    public Long totalOrders;

    @JsonProperty
    public Double totalValue;

    @JsonProperty
    public Double averageOrderValue;

    public ComplexOrderResult() {}

    public ComplexOrderResult(Integer userId, String userName, Long totalOrders,
                            Double totalValue, Double averageOrderValue) {
        this.userId = userId;
        this.userName = userName;
        this.totalOrders = totalOrders;
        this.totalValue = totalValue;
        this.averageOrderValue = averageOrderValue;
    }

    @Override
    public String toString() {
        return "ComplexOrderResult{" +
                "userId=" + userId +
                ", userName='" + userName + '\'' +
                ", totalOrders=" + totalOrders +
                ", totalValue=" + totalValue +
                ", averageOrderValue=" + averageOrderValue +
                '}';
    }
}
